### DISCUSSION TITLE
Storyboard shader support

### DISCUSSION THREAD

**Discussion by Tunnelbliick** [ppy/osu#24340] [Ideas]
Comming from someone that loves modcharts and games like NotITG i would love to have added shader support for storyboards, or just a way to add shaders ontop of the storyboard.

Ive been recently working away at mania storyboarding, and alot of things are already possible with stuff like storybrew and other community driven applications to make the writing of storyboard more accessable. Which you can already do alot of stuff with a bit of igenuity and knowing coding.

But in the end all we can do is move sprites around. We cant even properly transform them to give the illusion of pivot or tilting.

I think theres just a lot of value allowing shaders.

**Comment by Paladynee** [comment]
i dont think many people would benefit from this, when considering the amount of work this change needs, i think its very low priority.
and cross-renderer compatibility is a question here

**Reply by Tunnelbliick** [reply]
Storyboarding has been kinda pushed to the side for a while now, hence why anything storyboarding affects "almost" no one.
That doesnt mean that it woudnt be benefitial to the game as a whole. im pretty sure ppy mentioned before that he is more fond of the visually impressive play aka. BG or video on, which modcharts and other storyboard maps kinda focus on since you cannot play without them on.

Putting it aside and saying "just use videos" is also not rlly optimal since then we are limited by the framecount, which would be a whole different issue.

Its true that there is some issues between different renderers, but i dont think that there woudnt be a solution for it. Since we talking about having shaders written by users for storyboards u could ofload that issue to the users aswell and just have them forcefully write them for multiple renderers, if there rlly is no other way to convert them.

Its just sad that that all the other parts of the game get alot of improvements whilst storyboarding ever since is kinda beeing left in the corner, 3d already is unfinished so i doubt this would happen anyways.

**Reply by tadatomix** [reply]
> i dont think many people would benefit from this, when considering the amount of work this change needs, i think its very low priority. and cross-renderer compatibility is a question here

I think they would? I mean, it would be a great start. People can start writing shaders and sharing them on some websites, while others, if they need to, could use them in their storyboards. And even if we simplify it to small amount effects (see ADOFAI and Rhythm Doctor), it still would be better than what we have, since, if you look at what people can do with them in these games, it's impressive.

**Reply by tadatomix** [reply]
In case you need an example of how sharing shaders could work here's three examples

### Example №1: Website sharing

1. Storyboarder wants to implement a funky effect in storyboard
2. They go searching something like "osu storyboard shaders"
3. They're stumbling upon some hub with shaders (example: Godot Shaders)
4. In the gallery of many examples, they are finding one, which fits into their needs
5. They are pasting it into their code

### Example №2: Discord/Commissions

1. Storyboarder wants to implement a funky effect in storyboard
2. They go to some storyboard related server and ask for a shader
3. Somebody helps them and sending a code for this shader
4. They are pasting it into their code

### Example №3: Tutorials

1. Storyboarder wants to implement a funky effect in storyboard
2. They go to YouTube/osu!wiki and find a tutorial how to code shaders
3. After some time, they are nailing making shaders
4. Now their storyboards have beautiful looking effects

**Comment by bdach** [comment]
This is another one of those proposals that I don't think is going to happen before people meaningfully move off of stable, because this sort of feature will not be backported, so it will end up fragmenting the player base by having beatmaps that are only playable as intended on a client that 20% of users use.

lazer doesn't even have a design mode implemented because it will take months of effort for (in my opinion, at this time) disproportionate gain. Piling shader support on is premature.

**Reply by tadatomix** [reply]
That's a sad true, yet so it really would be good to see support of this in the future.
